webapp/packages/plugin-navigation-tree/src/NodesManager/NavNodeContextMenuService.ts

```diff
@@ -180,11 +181,21 @@ export class NavNodeContextMenuService extends Bootstrap {
       },
     });
 
+    this.menuService.setHandler({
+      id: 'menu-navigation-tree-create',
+      menus: [MENU_NAVIGATION_TREE_CREATE],
+      isHidden: context => {
+        const items = this.menuService.getMenuItemCreators(context);
+
+        return items.length === 0;
+      },
```

Comment:
menus is hidden by default if there is no applicable items as i know, so basically you don't need to manually handle this